components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -309,15 +315,32 @@ AdsServiceImpl::AdsServiceImpl(
 
   MigrateConfirmationState();
 
+  g_brave_browser_process->resource_component()->AddObserver(this);
+
   rewards_service_->AddObserver(this);
+
+  if (kShouldAlwaysRunService.Get()) {
+    RegisterResourceComponentsForDefaultLocale();
+  }
 }
 
 AdsServiceImpl::~AdsServiceImpl() {
+  g_brave_browser_process->resource_component()->RemoveObserver(this);
```

Comment:
this is a layering violation, you need to pass `g_brave_browser_process->resource_component()` to the constructor